verified_boot_app.md

```diff
@@ -11,8 +38,109 @@ Src: https://github.com/Kicksecure/kicksecure-wiki-backup/blob/6900fdfbdea8693fd
+     However, the architecture support MAY NOT be limited to AMD64 only and
+     MAY also work on any architecture supported by the UEFI Specification 2.7
+     (refer to Section 2.3 Calling Conventions).
+
+<!--
+* Vendor neutral: yes, any operating system that wishes to be compatible with the "EFI_ALT" standard/convention is welcome to.
+* Kicksecure specific: no
+* Kicksecure branding: no
+-->
+
+1.4. The application MUST be OS/HW vendor agnostic and NOT subject to any
+     branding.
+
+### Keys and boot options management
+
+<!--
+* Boot compatibility with mainstream operating systems such as Microsoft Windows: Yes, the user-controlled verified boot feature should not restrict the user's freedom to boot any operating system, if they so confirm in their firmware. (This is useful to avoid "Linux-only motherboard" and to keep resale value higher.)
+-->
+
+2.1. The application MUST NOT prevent a user from booting any UEFI-compliant
+     operating system. It MAY prompt the user to grant trust to an unknown key
+     or to disable UEFI Secure Boot for that boot, but it must always offer at
+     least one successful path.
+
+<!--
+* Support for binaries signed by multiple keys: Yes. (UEFI already supports that.) (This is important to allow the distribution or customer to cycle/replace the key in later upgrades or for planned signing key changes.)
+-->
+
+2.2. The application MUST support images singed by multiple keys.
+
+<!--
+* Public key TOFU (trust on first use): yes, if confirmed in the firmware.
+-->
+
+2.3. The application MUST ask the user for confirmation to trust given key
+     on the first use. The application MAY maintain the base of trusted keys
+     separately from UEFI Secure Boot or use existing UEFI Secure Boot
+     databases.
+
+<!--
+* Show key fingerprint: yes - "key fingerprint [key fingerprint]" [info symbol] "What is this? Brief explanation that the user can check with the vendor of this operating system if they wish to trust that key."
+-->
+
+2.4. The application SHOULD present the information about keys and
+     certificates in human-readable form using Issuer's and Subject's Common
+     Name (CN) and Organization (O) or other fields if present. Additionally,
+     the key fingerprint MUST always be displayed.
+
+<!--
+* Microsoft Secure Boot default signing key:
+** included in the firmware by default: yes, required for the purpose of allowing the user to boot Microsoft Windows
+** trusted in the firmware by default: no, express permission by the user is needed. (Because this key could lead to Secure Booting a Linux malware distribution. This would be unwanted by users who wish to use full verified boot with their chosen key.)
+* What this is:
+** A mechanism for users to much more easily choose which keys they trust.
+-->
+
+2.5. The firmware SHOULD include the Microsoft UEFI Secure Boot keys and
+     certificates in the firmware vendor's default set of UEFI Secure Boot
+     keys.
```

Comment:
I don't think we need this requirement, it is not related to application and passing or failing this requirements should not affect how application work.

Reply:
If there will be no bootloader signed by anything else than MS key, then the application would not be able to offer a bootable path.

Reply:
And that is also the case if there is no application, so there is no difference in behavior. We should not define requirements for firmware; we should define requirements for the application and handle various firmware states, even if MS keys do not exist in the system.

Reply:
Yes, but if there is no application, that's not our concern.

Also the application will have to fall back to setup in case no bootable option is present.